src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionFactory.cs

```diff
@@ -825,6 +825,23 @@ static bool IsAsyncMethod(MethodInfo method)
                     {
                         try
                         {
+                            if (value is string potentiallyJsonString)
+                            {
+                                Debug.Assert(typeInfo.Type != typeof(string), "string parameters should not enter this branch.");
+
+                                // Account for the parameter potentially being a JSON string.
+                                // This is needed for compatibility with Semantic Kernel,
+                                // which may pass parameters as JSON strings.
```

Comment:
Is this something that can/should be fixed at the SK layer instead?

Or if what SK is doing is appropriate, then should this comment be tweaked? At that point this wouldn't be about SK compat and would instead be about supporting such usage in general, regardless of the consumer who at that point is doing something reasonable by definition.

Reply:
Possibly? I went for this approach because we already do something similar reconciling JsonElement and JsonNode inputs. The fact we're explicitly not entering this path for arguments of type string (something that SK is also doing) makes me less worried about ambiguity.